analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/SnippetCompiler.cs

```diff
@@ -154,14 +154,15 @@ public IPropertySymbol GetPropertySymbol(string typeDotMethodName)
             new SyntaxNodeAnalysisContext(node, SemanticModel, null, null, null, CancellationToken.None);
 
         private static bool HasCompilationErrors(Compilation compilation) =>
-            compilation.GetDiagnostics().Any(x => x.Severity == DiagnosticSeverity.Error && (x.Id.StartsWith("CS") || x.Id.StartsWith("BC")));
+            compilation.GetDiagnostics().Any(IsCompilationError);
+
+        private static bool IsCompilationError(Diagnostic diagnostic) =>
+            diagnostic.Severity == DiagnosticSeverity.Error && (diagnostic.Id.StartsWith("CS") || diagnostic.Id.StartsWith("BC"));
 
         private static void DumpCompilationErrors(Compilation compilation)
         {
-            var diagnostics = compilation.GetDiagnostics();
-
             Console.WriteLine("Diagnostic errors:");
-            foreach (var d in diagnostics)
+            foreach (var d in compilation.GetDiagnostics().Where(IsCompilationError))
```

Comment:
_[random thought, not actionable]_ when I see this change I realize (as a code reviewer) that having tests for the test framework would be nice in the context of this PR

Reply:
It actually is actionable. We do have some UTs for some parts of the TestFramework. I can at least bootstrap the class to have at least something.